lib/wallet/src/Cardano/Wallet/DB.hs

```diff
@@ -332,6 +343,308 @@ data DBLayer m s k = forall stm. (MonadIO stm, MonadFail stm) => DBLayer
+    -- and distributed the smaller layer parts as well.
+    , rollbackTo_
+        :: WalletId
+        -> Slot
+        -> ExceptT ErrNoSuchWallet stm ChainPoint
+    , prune_
+        :: WalletId
+        -> Quantity "block" Word32
+        -> ExceptT ErrNoSuchWallet stm ()
+    , atomically_
+        :: forall a. stm a -> m a
+    }
+
+-- | Create a legacy 'DBLayer' from smaller database layers.
+mkDBLayerFromParts
+    :: (MonadIO stm, MonadFail stm)
+    => DBLayerCollection stm m s k -> DBLayer m s k
+mkDBLayerFromParts DBLayerCollection{..} = DBLayer
+    { initializeWallet = initializeWallet_ dbWallets
+    , removeWallet = removeWallet_ dbWallets
+    , listWallets = listWallets_ dbWallets
+    , walletsDB = walletsDB_ dbCheckpoints
+    , putCheckpoint = putCheckpoint_ dbCheckpoints
+    , readCheckpoint = readCheckpoint_ dbCheckpoints
+    , listCheckpoints = listCheckpoints_ dbCheckpoints
+    , putWalletMeta = putWalletMeta_ dbWalletMeta
+    , readWalletMeta = readWalletMeta_ dbWalletMeta
+    , isStakeKeyRegistered = isStakeKeyRegistered_ dbDelegation
+    , putDelegationCertificate = putDelegationCertificate_ dbDelegation
+    , putDelegationRewardBalance = putDelegationRewardBalance_ dbDelegation
+    , readDelegationRewardBalance = readDelegationRewardBalance_ dbDelegation
+    , putTxHistory = putTxHistory_ dbTxHistory
+    , readTxHistory = readTxHistory_ dbTxHistory
+    , getTx = getTx_ dbTxHistory
+    , putLocalTxSubmission = putLocalTxSubmission_ dbPendingTxs
+    , readLocalTxSubmissionPending = readLocalTxSubmissionPending_ dbPendingTxs
+    , updatePendingTxForExpiry = updatePendingTxForExpiry_ dbPendingTxs
+    , removePendingOrExpiredTx = removePendingOrExpiredTx_ dbPendingTxs
+    , putPrivateKey = putPrivateKey_ dbPrivateKey
+    , readPrivateKey = readPrivateKey_ dbPrivateKey
+    , readGenesisParameters = readGenesisParameters_ dbWallets
+    , rollbackTo = rollbackTo_
+    , prune = prune_
+    , atomically = atomically_
+    }
+
+-- | A database layer for a collection of wallets
+data DBWallets stm s = DBWallets
+    { initializeWallet_
+        :: WalletId
+        -> Wallet s
+        -> WalletMetadata
+        -> [(Tx, TxMeta)]
+        -> GenesisParameters
+        -> ExceptT ErrWalletAlreadyExists stm ()
+        -- ^ Initialize a database entry for a given wallet. 'putCheckpoint',
+        -- 'putWalletMeta', 'putTxHistory' or 'putProtocolParameters' will
+        -- actually all fail if they are called _first_ on a wallet.
+
+    , readGenesisParameters_
+        :: WalletId
+        -> stm (Maybe GenesisParameters)
+        -- ^ Read the *Byron* genesis parameters.
+
+    , removeWallet_
+        :: WalletId
+        -> ExceptT ErrNoSuchWallet stm ()
+        -- ^ Remove a given wallet and all its associated data (checkpoints,
+        -- metadata, tx history ...)
+
+    , listWallets_
+        :: stm [WalletId]
+        -- ^ Get the list of all known wallets in the DB, possibly empty.
```

Comment:
Personally I prefer to avoid comments like this one as it doesn't add any information that's not already present in the code.